sdks/java/extensions/smb/src/test/java/org/apache/beam/sdk/extensions/smb/MixedSourcesEndToEndTest.java

```diff
@@ -54,11 +60,11 @@
   @Rule public final transient TestPipeline pipeline = TestPipeline.create();
   @Rule public final transient TestPipeline pipeline2 = TestPipeline.create();
   @Rule public final transient TestPipeline pipeline3 = TestPipeline.create();
-  @Rule public final TemporaryFolder source1Folder = new TemporaryFolder();
-  @Rule public final TemporaryFolder source2Folder = new TemporaryFolder();
+  @Rule public final transient TemporaryFolder source1Folder = new TemporaryFolder();
```

Comment:
why do these need to be transient? come to think of it why did I even make the `TestPipeline`s transient to begin with? 😅

Reply:
Otherwise they get pulled into the anonymous `DoFn`. There might be other ways to fix this. I'll try some ideas.

Reply:
Right it works if I make the `DoFn` a static class. Also `transient` is not necessary for the `TestPipeline`s.